test/integration_test/common_test.go

```diff
@@ -70,11 +72,45 @@ func validateStorageCluster(
 	cluster *corev1alpha1.StorageCluster,
 	timeout, interval time.Duration,
 ) error {
-	// Validate expected portworx pods are running and no more, no less
-	return validateStorageClusterPods(cluster, timeout, interval)
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if err = validateStorageClusterPods(cluster, timeout, interval); err != nil {
+		return err
+	}
+
+	svc, err := k8s.Instance().GetService("portworx-service", cluster.Namespace)
+	pxEndpoint := ""
+	if err == nil {
+		pxEndpoint = svc.Spec.ClusterIP
+	}
+
+	conn, err := grpc.Dial(pxEndpoint, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	nodeClient := api.NewOpenStorageNodeClient(conn)
+	nodeEnumerateResp, err := nodeClient.Enumerate(context.Background(), &api.SdkNodeEnumerateRequest{})
+	if err != nil {
+		return err
+	}
+	expectedNodes := len(cluster.Spec.Nodes)
```

Comment:
The cluster nodes are only used to overwrite the node specific spec, so it may not always be populated. If there are no nodes in this section, it means that all nodes will use the configuration given at the cluster.Spec level.

We should validate the number of px nodes should be equal to `expectedPods()`. This method takes into consideration node affinity rules if there are any.